pkg/tree/schema.ipldsch

```diff
@@ -0,0 +1,41 @@
+type ProllyNode struct {
+    # true for leaf nodes and false for branch nodes
+    isLeaf Bool
+    # raw keys(input from users) for leaf node. For branch node, the key is last key in the child node
+	keys [Bytes]
+	# input data from users for leaf nodes, link(cid) for branch nodes
+    values [Any]
+} representation tuple
+
+type ProllyRoot struct{
+    # root ProllyNode of tree
+    RootCid &ProllyNode
+    # config about the tree
+    ConfigCid &TreeConfig
+}
+
+type SuffixThresholdConfig struct {
+    chunkingFactor Int
+} representation tuple
+
+# Tree Config for prolly tree, it includes some global setting, the splitter method you choose and specific configs about
+# the splitter
+type TreeConfig struct{
+    strategyType    Int
+    minNodeSize     Int
+    maxNodeSize     Int
+    maxPairsInNode  Int
+    nodeCodec       Prefix
+    strategy        strategy
+} representation tuple
```

Comment:
Please use the same ordering of keys as in the spec code. MaxSize, MinSize, cidVersion, codec, hashFunction, nullable hashLength, strategy ChunkingStrategy

Reply:
maybe we should strategyType(Int) to specify the strategy type? It makes code easy to get the specific strategy, because the strategy in Go is like below, we will check every strategy without type 
type strategy struct {
	//Weilbull    *WeibullThresholdConfig
	//RollingHash *RollingHashConfig
	Suffix *PrefixThresholdConfig
}

Reply:
Ah, does Go not have a way to say "This field is one of three types"?

Reply:
As I know, there isn't convenient way to get the real config from the union struct in Go

Reply:
We should get rid of the union and split it into multiple keys in the spec, then.